src/Components/Server/src/Circuits/RemoteRenderer.cs

```diff
@@ -79,6 +83,44 @@ public Task AddComponentAsync(Type componentType, string domElementSelector)
             return RenderRootComponentAsync(componentId);
         }
 
+        protected override void ProcessRenderQueue()
+        {
+            if (_unacknowledgedRenderBatches.Count >= _options.MaxBufferedUnacknowledgedRenderBatches)
+            {
+                // If we got here it means we are at max capacity, so we don't want to actually process the queue,
+                // as we have a client that is not acknowledging render batches fast enough (something we consider needs
+                // to be fast).
+                // The result is somethign as follows:
+                // Lets imagine an extreme case where the server produces a new batch every milisecond.
+                // Lets say the client is able to ACK a batch every 100 miliseconds.
+                // When the app starts the client might see the sequence 0.000->0.{MAXUnacknowledgeRenderBatches} and then
```

Comment:
Let me know if I'm misinterpreting, but this was hard for me to understand. Is the following correct?

```suggestion
                // When the app starts the client might see the sequence 0->(MaxUnacknowledgedRenderBatches-1) and then
```